File: Hash_Table/560.Subarray_Sum_Equals_K/Subarray_Sum_Equals_K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        int ans = 0, curr = 0;
        mp[0]++;
        for(int i: nums){
            curr += i;
            ans += mp[curr - k];
            mp[curr]++;
        }
        return ans;
    }
};

File: Hash_Table/560.Subarray_Sum_Equals_K/Subarray_Sum_Equals_K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hash_map = defaultdict(int)
        curr = 0
        ans = 0
        hash_map[0] = 1
        for i in nums:
            curr += i
            ans += hash_map[curr - k]
            hash_map[curr] += 1
        return ans